webpack.config.js

```diff
@@ -181,33 +156,59 @@ module.exports = ( env ) => {
 
     ]),
     performance: {
-      hints: ifTest() ? false : 'warning'
+      hints: ifProd() && 'warning'
     }
   }
 
 
-  function ifProdOrSite( callback = true ) {
+  function ifProdOrSite( returnValue = true ) {
```

Comment:
ok now I see what you try to do in here - mimic `getIfUtils`.
So we should adhere to their implementation and make things easier. 

```js
function getCustomIfUtils({ifDev, ifProd, ifSite}={}){
   return {
     ifProdOrSite(value, alternate){  return getByEnvValue(ifProd() || isSite(), value, alternate )  },
     ifDevOrSite(value, alternate){ return getByEnvValue(ifDev() || isSite(), value, alternate ) }
   }

   function getByEnvValue(envValue, value, alternate ){
     return isUndefined(value) ? envValue : propIf(envValue, value, alternate )
   }

   function isUndefined(val){ return typeof val === "undefined" }
   function getValue(val) {
    return JSON.parse(val)
   }
   function propIf(add, value, alternate) {
     return getValue(add) ? value : alternate
   }
}
```

and then on the top

```js
module.exports = ( env ) => {
  const { ifProd, ifNotProd, ifTest, ifDev, ifSite } = getIfUtils( env, ['prod', 'test', 'dev', 'site'] );
  const { ifProdOrSite, ifDevOrSite } = getCustomIfUtils({ifDev,ifProd,ifSite});
  // ...
}
```